src/kibana/plugins/settings/sections/indices/_indexed_fields.js

```diff
@@ -26,11 +27,16 @@ define(function (require) {
           { title: 'controls', sortable: false }
         ];
 
-        $scope.$watchCollection('indexPattern.fields', function () {
+        $scope.$watchCollection('indexPattern.fields', refreshRows);
+        $scope.$watch('fieldFilter', refreshRows);
```

Comment:
Using multi watch here would be better, so the handler is only called once when instantiated. IIRC, something like `$scope.$watchMulti(['fieldFilter', '[]indexPattern.fields'], refreshRows);`, where `[]` is used to watch a collection - double-check that syntax though.